From 536a020e68c000298cbc88ba5c2c4acf636d564f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:37 -0400
Subject: [usb] xhci: remove the unused ->address field

Message-id: <1410887463-20674-86-git-send-email-dzickus@redhat.com>
Patchwork-id: 95271
O-Subject: [RHEL7 PATCH 085/591] usb: xhci: remove the unused ->address field
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit a2cdc3432c361bb885476d1c625e22b518e0bc07
Author: Dan Williams <dan.j.williams@intel.com>
Date:   Wed Oct 16 12:25:44 2013 -0700

    usb: xhci: remove the unused ->address field

    Only used for debug output, so we don't need to save it.

    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 9ecd82a..4265b48 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -3716,9 +3716,6 @@ disable_slot:
  * the device).
  * We should be protected by the usb_address0_mutex in khubd's hub_port_init, so
  * we should only issue and wait on one address command at the same time.
- *
- * We add one to the device address issued by the hardware because the USB core
- * uses address 1 for the root hubs (even though they're not really devices).
  */
 int xhci_address_device(struct usb_hcd *hcd, struct usb_device *udev)
 {
@@ -3863,16 +3860,13 @@ int xhci_address_device(struct usb_hcd *hcd, struct usb_device *udev)
  slot_ctx = xhci_get_slot_ctx(xhci, virt_dev->out_ctx);
  trace_xhci_address_ctx(xhci, virt_dev->out_ctx,
     slot_ctx->dev_info >> 27);
- /* Use kernel assigned address for devices; store xHC assigned
-  * address locally. */
- virt_dev->address = (le32_to_cpu(slot_ctx->dev_state) & DEV_ADDR_MASK)
-  + 1;
  /* Zero the input context control for later use */
  ctrl_ctx->add_flags = 0;
  ctrl_ctx->drop_flags = 0;
 
  xhci_dbg_trace(xhci, trace_xhci_dbg_address,
-   "Internal device address = %d", virt_dev->address);
+         "Internal device address = %d",
+         le32_to_cpu(slot_ctx->dev_state) & DEV_ADDR_MASK);
 
  return 0;
 }
diff --git a/drivers/usb/host/xhci.h b/drivers/usb/host/xhci.h
index ed3a425..03c74b7 100644
--- a/drivers/usb/host/xhci.h
+++ b/drivers/usb/host/xhci.h
@@ -935,8 +935,6 @@ struct xhci_virt_device {
  /* Rings saved to ensure old alt settings can be re-instated */
  struct xhci_ring  **ring_cache;
  int    num_rings_cached;
- /* Store xHC assigned device address */
- int    address;
 #define XHCI_MAX_RINGS_CACHED 31
  struct xhci_virt_ep  eps[31];
  struct completion  cmd_completion;
-- 
1.7.1